pkg/endpoint/metrics.go

```diff
@@ -96,7 +95,7 @@ func (s *regenerationStatistics) GetMap() map[string]*spanstat.SpanStat {
 		"proxyWaitForAck":        &s.proxyWaitForAck,
 		"mapSync":                &s.mapSync,
 		"prepareBuild":           &s.prepareBuild,
-		logfields.BuildDuration:  &s.totalTime,
+		"total":                  &s.totalTime,
```

Comment:
just curious if we need to keep old metric and mark it as deprecated (i.e. will be removed in 1.10) ?

Reply:
> One small question. Can you also help to update to new metric for a bunch of dashboards we have in repo as well ?

Absolutely, thanks for the pointer!

> just curious if we need to keep old metric

Possibly, what is our deprecation policy regarding these kind of changes?

Reply:
>Possibly, what is our deprecation policy regarding these kind of changes?

I am not 100% sure. I just take a look at [docs](https://github.com/cilium/cilium/blob/master/Documentation/install/upgrade.rst) as mentioned in below comment, there are a bunch `new metrics` and `deprecated` metrics in this dev cycle, so I assume the common approach is `rename` + `deprecate` for any customer facing metric. /cc @joestringer @aanm

Reply:
I've added this as a user-facing change to `upgrade.rst`, but I'm on the fence about duplicating the metric. Whether we rename or duplicate it, things will break regardless.

If we duplicate, `sum(foo{scope!="buildDuration"})...` will now quietly sneak a `total` scope into the sum, inflating the value, but positive queries for `scope="buildDuration"` will continue to function like before. Obvious breakage will occur in 1.11.

If we rename, both positive and negative queries will break, the breakage will be clear, and will occur on the version where it's documented in the release notes.

As an operator, I prefer things clearly breaking during/after an upgrade instead of getting paged when AFK.